src/server/service/kv.rs

```diff
@@ -122,11 +122,18 @@ impl<T: RaftStoreRouter + 'static, E: Engine> tikvpb_grpc::Tikv for Service<T, E
         options.key_only = req.get_key_only();
         options.reverse_scan = req.get_reverse();
 
+        let end_key = if req.get_end_key().is_empty() {
```

Comment:
do we need to check "has_end_key()" to become backward compatible?

Reply:
It seems we don't have that method...?
The end_key should be empty if the field is not provided, I think...

Reply:
I tried to run this TiKV with old version of TiDB and there's no trouble about scanning

Reply:
Are you approving? I'll merge it..

Reply:
Ok, seems to be fine.